pmd-java/src/main/resources/category/java/security.xml

```diff
@@ -60,6 +60,34 @@ public class Foo {
         byte[] iv = "secret iv in here".getBytes();
     }
 }
+]]>
+        </example>
+    </rule>
+
+    <rule name="StaticFieldMustBeFinal"
+          language="java"
+          since="6.34.0"
+          message="Do not use non-final visible static fields"
+          class="net.sourceforge.pmd.lang.rule.XPathRule">
+        <description>
+            Visible static fields must be made constants (or immutable
```

Comment:
This description should be explicit about what a "visible" static field is. The rule takes it to mean "non-private". Some other rules also whitelist package-private members, so this needs to be clarified.

Reply:
I think, it would also be good to add a reference to the rule [AssignmentToNonFinalStatic](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#assignmenttononfinalstatic). Both rules are similar, but highlight different places: This rule highlights the static field itself, while AssignmentToNonFinalStatic highlights write access (assignments) to the static field.

I think, if you have "StaticFieldMustBeFinal" in your ruleset, then you don't need AssignmentToNonFinalStatic...